agent/src/beerocks/slave/backhaul_manager/backhaul_manager_thread.cpp

```diff
@@ -4319,6 +4328,154 @@ bool backhaul_manager::add_link_metrics(const sMacAddr &reporter_al_mac,
+                ret = false;
+            }
+            //TODO: find the way for copy tlvChannelPreference from rx to tx
+            auto channel_preference = cmdu_tx.addClass<wfa_map::tlvChannelPreference>();
+            if (!channel_preference) {
+                LOG(ERROR) << "Failed addClass<wfa_map::tlvChannelPreference>";
+                ret = false;
+            }
+            channel_preference->radio_uid() = mac;
+
+            auto op_class_channels_tx = channel_preference->create_operating_classes_list();
+            if (!op_class_channels_tx) {
+                LOG(ERROR) << "create_operating_classes_list() has failed!";
+                return false;
+            }
+
+            if (!op_class_channels_tx->alloc_channel_list()) {
+                LOG(ERROR) << "alloc_channel_list() has failed!";
+                return false;
+            }
+
+            auto power_limit_tlv = cmdu_tx.addClass<wfa_map::tlvTransmitPowerLimit>();
+            if (!power_limit_tlv) {
+                LOG(ERROR) << "Failed addClass<wfa_map::tlvTransmitPowerLimit>";
+                ret = false;
+            }
+
+            power_limit_tlv->radio_uid() = mac;
+
+            for (auto &power : transmit_power) {
+                if (std::get<0>(power) != mac) {
+                    continue;
+                }
+                power_limit_tlv->transmit_power_limit_dbm() = std::get<1>(power);
+            }
+
+            //if (!message_com::send_cmdu(socket->slave, cmdu_tx)) {
+            //    LOG(ERROR)
+            //        << "Failed forwarding CHANNEL_SELECTION_REQUEST_MESSAGE message to son_slave";
+            //    ret = false;
+            //    continue;
+            //}
+
+            m_channel_selection_request.push_back(mac);
+            ret = true;
+            break;
+        }
+    }
+
+    return ret;
+}
+
+bool backhaul_manager::handle_slave_channel_selection_response(ieee1905_1::CmduMessageRx &cmdu_rx,
+                                                               const std::string &src_mac)
+{
+    const auto mid = cmdu_rx.getMessageId();
+    LOG(DEBUG) << "Received CHANNEL_SELECTION_RESPONSE message, mid=" << std::hex << int(mid);
+
+    auto channel_selection_response = cmdu_rx.getClass<wfa_map::tlvChannelSelectionResponse>();
+    if (!channel_selection_response) {
+        LOG(ERROR) << "Failed cmdu_rx.getClass<wfa_map::tlvChannelSelectionResponse>(), mid="
+                   << std::hex << int(mid);
+        return false;
+    }
+
+    //LOG(DEBUG) << "VT channel_selection_response->radio_uid(): "
+    //           << tlvf::mac_to_string(channel_selection_response->radio_uid());
+    //LOG(DEBUG) << "VT channel_selection_response->response_code(): "
+    //           << int(channel_selection_response->response_code());
+
+    m_channel_selection_response.push_back({channel_selection_response->radio_uid(),
+                                            uint8_t(channel_selection_response->response_code())});
```

Comment:
(nitpick) Thanks to @mariomaz, I discovered that we can use emplace_back instead of push_back. It avoids an additional copy.

Reply:
emphase_back does not work for the marked case.
It will be work if i add 3 useless strings for creating and filling structure elements, then structure object i will be able to put inside the vector via amphase_back